block-layer/codecs/DAG-JSON.md

```diff
@@ -14,14 +14,56 @@ Contrary to popular belief, JSON as a format supports Big Integers. It's only
 JavaScript itself that has trouble with them. This means JS implementations
 of `dag-json` can't use the native JSON parser and serializer.
 
+### Version 0
+
+This is an old version of `dag-json` that reserved the `"/"` key in order to
+represent binary and link data types.
+
 #### Binary Type
 
 ```javascript
 {"/": { "base64": String }}
 ```
 
-### Link Type
+#### Link Type
 
 ```javascript
 {"/": String /* base encoded CID */}
-```
\ No newline at end of file
+```
+
+### Version 1
+
+#### Format
+
+The internal data format is valid JSON but is **NOT** identical to the decoded
+node data codecs produce.
+
+Example internal format:
+
+```javascript
+{ "/": { "_": 1 },
+  "data": { "hello": "world", { "obj": { "array": [0, 0] } } },
```

Comment:
So those `[0, 0]` are placeholders? So if you want to encode `[CID, "some string"]` it would be `[0, "some string"]`?

Reply:
They are just placeholders to preserve the ordering of the array.

Reply:
I'd use `null` for that, but that's a minor detail. Perhaps the example could be expended to `[null, "some string", null]` to make it clearer that there are place holders which are mixed with actual values.

Reply:
originally it was null but i went with 0 because null encoding is extra bytes :/